include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp

```diff
@@ -463,6 +463,13 @@ TimeIntBDF<dim, Number>::postprocessing() const
     operator_base->move_grid_and_update_dependent_data_structures(this->get_time());
   }
 
+  // We need to distribute the dofs for HDIV before computing the error since
+  // dealii::VectorTools::integrate_difference() does not take the periodic boundary
+  // constraints into account like MatrixFree does, hence reading the wrong value.
+  // distribute() updates the constrained values.
+  if(operator_base->get_spatial_discretization() == SpatialDiscretization::HDIV)
+    operator_base->get_constraint_u().distribute(const_cast<VectorType &>(get_velocity(0)));
+
```

Comment:
Yes, I agree that it doesn't really fit in here, but I think there will be problems below in the else branch when calculating the error, i.e lines 491-492. So if `standard` is ever changed to false... I can add a comment.

Reply:
I think you would need to call distribute before this line:
https://github.com/exadg/exadg/blob/d405d108cc1231ae7333a1fe2e631b7a8a58dcfb/include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp#L491

Reply:
How do we handle inhomogeneous Dirichlet boundaries (or periodic boundaries) in the structure module? I think we should try to apply similar techniques, as those are needed for all forms of constrained unknowns. In particular, we should also anticipate the case of adaptive mesh refinement, where the velocity field will be subject to hanging node constraints. I did have a short look, but could not find it immediately (but I remember we discussed it at some point, as there are several solutions to the homogeneous/inhomogeneous representation needed at various stages). This does not necessarily have to happen in this PR, depending on how precisely we are going to solve it.

Reply:
We should remove the else branch anyway from the time integrator, since this is not implemented in a nice way. It has previously been used to test certain things but should better be realized in the postprocessor. This is something I can address in the future.

@kronbichler What I did not understand is whether the aspects you mention are important for obtaining the correct solution, or just for postprocessing? I understood @NiklasWik in the sense that the solution is computed correctly and this discussion is only about doing the postprocessing correctly. Since Niklas has finished his project and will not implemented everything related to inhomogeneous BCs, we should find a solution that allows to complete this PR soon. If you address more general aspects here, could I ask you to summarize them in a separate issue such that they do not get lost and can be addressed later?